### PR TITLE
Render particles to render surface

### PR DESCRIPTION
**Pull request by LarsIngo**

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 10 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ParticleManager.cpp#L60](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c26050d9d73132fc73b41b99a5d1966c4656da4a/src/Engine/Manager/ParticleManager.cpp#L60): The function 'UpdateBuffer' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L325](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c26050d9d73132fc73b41b99a5d1966c4656da4a/src/Engine/Manager/RenderManager.cpp#L325): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L335](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c26050d9d73132fc73b41b99a5d1966c4656da4a/src/Engine/Manager/RenderManager.cpp#L335): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L353](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c26050d9d73132fc73b41b99a5d1966c4656da4a/src/Engine/Manager/RenderManager.cpp#L353): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L372](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c26050d9d73132fc73b41b99a5d1966c4656da4a/src/Engine/Manager/RenderManager.cpp#L372): The function 'GetLenses' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L392](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c26050d9d73132fc73b41b99a5d1966c4656da4a/src/Engine/Manager/RenderManager.cpp#L392): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L410](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c26050d9d73132fc73b41b99a5d1966c4656da4a/src/Engine/Manager/RenderManager.cpp#L410): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L429](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c26050d9d73132fc73b41b99a5d1966c4656da4a/src/Engine/Manager/RenderManager.cpp#L429): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L450](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c26050d9d73132fc73b41b99a5d1966c4656da4a/src/Engine/Manager/RenderManager.cpp#L450): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Manager/RenderManager.cpp [hunk 1]
```diff
@@ -89,6 +89,8 @@ void RenderManager::Render(World& world, bool soundSources, bool particleEmitter
             { GPUPROFILE("Render main window", Video::Query::Type::TIME_ELAPSED);
                 const glm::mat4 projectionMatrix = camera->GetComponent<Lens>()->GetProjection(mainWindowRenderSurface->GetSize());
                 const glm::mat4 viewMatrix = glm::inverse(camera->GetModelMatrix());
+                const glm::vec3 position = camera->GetWorldPosition();
+                const glm::vec3 up(viewMatrix[0][1], viewMatrix[1][1], viewMatrix[2][1]);
```
Comment (SwedishSock) [inline src/Engine/Manager/RenderManager.cpp]: Can be changed into up = camera->GetWorldOrientation() * glm::vec3(0, 1, 0); 
Once the quaternion refactoring branch is merged

File: src/Engine/Manager/RenderManager.cpp [hunk 2]
```diff
@@ -138,6 +147,8 @@ void RenderManager::Render(World& world, bool soundSources, bool particleEmitter
                     const glm::mat4 eyeTranslation = Managers().vrManager->GetHMDHeadToEyeMatrix(nEye);
                     const glm::mat4 eyeViewMatrix = eyeTranslation * lensViewMatrix;
                     const glm::mat4 projectionMatrix = headset->GetHMDProjectionMatrix(nEye, lens->zNear, lens->zFar);
+                    const glm::vec3 position = camera->GetWorldPosition();
+                    const glm::vec3 up(lensViewMatrix[0][1], lensViewMatrix[1][1], lensViewMatrix[2][1]);
```
Comment (SwedishSock) [inline src/Engine/Manager/RenderManager.cpp]: Same here with up vector

File: src/Engine/Manager/RenderManager.cpp (hunk 1 above)
Comment (LarsIngo) [reply src/Engine/Manager/RenderManager.cpp]: const glm::vec3 up(viewMatrix[0][1], viewMatrix[1][1], viewMatrix[2][1]); is faster

File: src/Engine/Manager/RenderManager.cpp (hunk 2 above)
Comment (LarsIngo) [reply src/Engine/Manager/RenderManager.cpp]: const glm::vec3 up(viewMatrix[0][1], viewMatrix[1][1], viewMatrix[2][1]); is faster

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 10 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [RenderManager.cpp#L342](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2ef1d6c376786daf09ab23299b1589a9269422cb/src/Engine/Manager/RenderManager.cpp#L342): Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L352](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2ef1d6c376786daf09ab23299b1589a9269422cb/src/Engine/Manager/RenderManager.cpp#L352): The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L370](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2ef1d6c376786daf09ab23299b1589a9269422cb/src/Engine/Manager/RenderManager.cpp#L370): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L389](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2ef1d6c376786daf09ab23299b1589a9269422cb/src/Engine/Manager/RenderManager.cpp#L389): The function 'GetLenses' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L409](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2ef1d6c376786daf09ab23299b1589a9269422cb/src/Engine/Manager/RenderManager.cpp#L409): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L427](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2ef1d6c376786daf09ab23299b1589a9269422cb/src/Engine/Manager/RenderManager.cpp#L427): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L446](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2ef1d6c376786daf09ab23299b1589a9269422cb/src/Engine/Manager/RenderManager.cpp#L446): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L467](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2ef1d6c376786daf09ab23299b1589a9269422cb/src/Engine/Manager/RenderManager.cpp#L467): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [EntityCheck.cpp#L4](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2ef1d6c376786daf09ab23299b1589a9269422cb/src/Tests/engine/EntityCheck.cpp#L4): The function 'TEST_CASE' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
iamheman

**Pull request merged**